gcloud-java-compute/src/test/java/com/google/gcloud/compute/ComputeImplTest.java

```diff
@@ -1907,6 +1928,148 @@ public void testCreateRegionAddressWithOptions() {
   }
 
   @Test
+  public void testCreateSnapshot() {
+    EasyMock.expect(computeRpcMock.createSnapshot(DISK_ID.zone(), DISK_ID.disk(),
+        SNAPSHOT_ID.snapshot(), null, EMPTY_RPC_OPTIONS)).andReturn(globalOperation.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Operation operation = compute.create(SNAPSHOT);
+    assertEquals(globalOperation, operation);
+  }
+
+  @Test
+  public void testCreateSnapshotWithOptions() {
+    Capture<Map<ComputeRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    EasyMock.expect(computeRpcMock.createSnapshot(eq(DISK_ID.zone()), eq(DISK_ID.disk()),
+        eq(SNAPSHOT_ID.snapshot()), EasyMock.<String>isNull(), capture(capturedOptions)))
+            .andReturn(globalOperation.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Operation operation = compute.create(SNAPSHOT, OPERATION_OPTION_FIELDS);
+    String selector = (String) capturedOptions.getValue().get(OPERATION_OPTION_FIELDS.rpcOption());
+    assertTrue(selector.contains("selfLink"));
+    assertTrue(selector.contains("id"));
+    assertTrue(selector.contains("description"));
+    assertEquals(23, selector.length());
+    assertEquals(globalOperation, operation);
+  }
+
+  @Test
+  public void testGetSnapshot() {
+    EasyMock.expect(computeRpcMock.getSnapshot(SNAPSHOT_ID.snapshot(), EMPTY_RPC_OPTIONS))
+        .andReturn(SNAPSHOT.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Snapshot snapshot = compute.getSnapshot(SNAPSHOT_ID.snapshot());
+    assertEquals(new Snapshot(compute, new SnapshotInfo.BuilderImpl(SNAPSHOT)), snapshot);
+  }
+
+  @Test
+  public void testGetSnapshot_Null() {
+    EasyMock.expect(computeRpcMock.getSnapshot(SNAPSHOT_ID.snapshot(), EMPTY_RPC_OPTIONS))
+        .andReturn(null);
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    assertNull(compute.getSnapshot(SNAPSHOT_ID.snapshot()));
+  }
+
+  @Test
+  public void testGetSnapshotWithSelectedFields() {
+    Capture<Map<ComputeRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    EasyMock.expect(computeRpcMock.getSnapshot(eq(SNAPSHOT_ID.snapshot()),
+        capture(capturedOptions))).andReturn(SNAPSHOT.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Snapshot snapshot = compute.getSnapshot(SNAPSHOT_ID.snapshot(), SNAPSHOT_OPTION_FIELDS);
+    String selector = (String) capturedOptions.getValue().get(SNAPSHOT_OPTION_FIELDS.rpcOption());
+    assertTrue(selector.contains("selfLink"));
+    assertTrue(selector.contains("id"));
+    assertTrue(selector.contains("description"));
+    assertEquals(23, selector.length());
+    assertEquals(new Snapshot(compute, new SnapshotInfo.BuilderImpl(SNAPSHOT)), snapshot);
+  }
+
+  @Test
+  public void testDeleteSnapshot_Operation() {
```

Comment:
Nit: in one of the delete snapshot tests, you could try using the string name of the snapshot instead of SNAPSHOT_ID (to cover that method in the tests).